### PR TITLE
boards/meshme: Add ADC support and tests

### PR DESCRIPTION
**Pull request by luisan00**
### Contribution description
This PR contains two **related** commits:
-  Add support for ADC peripheral on two lines of the `meshme` platform
-  `tests/periph_adc`

### Testing procedure
Run the `periph_adc` test with:
```sh
make BOARD=meshme flash test
```

### Issues/PRs references
None

### CODE REVIEW

**Review by eduazocar — CHANGES_REQUESTED** [review]
This looks good, only need to fix a little EOF, to merge this

File: boards/meshme/include/periph_conf.h [hunk 1]
```diff
@@ -29,17 +28,59 @@ extern "C" {
 
 #include "cpu.h"
 #include "periph_cpu.h"
-#include "periph/cpuid.h"
+//#include "periph/cpuid.h"
```
Comment (eduazocar) [inline boards/meshme/include/periph_conf.h:31]: dead code

File: tests/periph_adc/doc.txt [hunk 2]
```diff
@@ -0,0 +1,15 @@
+/**
+@defgroup      periph_adc
+@ingroup       test_group
+@{
+
+## Test the [A]nalog to [D]igital [C}onverter peripheral
```
Comment (eduazocar) [inline tests/periph_adc/doc.txt:6]:
```suggestion
## Test the [A]nalog to [D]igital [C]onverter peripheral
```

**Review by eduazocar — CHANGES_REQUESTED** [review]

File: tests/periph_adc/app.config.test [hunk 3]
```diff
@@ -0,0 +1,10 @@
+# this file enables modules defined in Kconfig. Do not use this file for
+# application configuration. This is only needed during migration.
+# enable ADC peripheral driver
+CONFIG_MODULE_PERIPH_ADC=y
+
+# enable xtimer to wake up periodically
+CONFIG_MODULE_ZTIMER=y
+CONFIG_MODULE_ZTIMER_MSEC=y
+
+CONFIG_MODULE_EMBUNIT=y
```
[the remark this answers is not in the document]
Comment (luisan00) [reply tests/periph_adc/app.config.test]: done

File: boards/meshme/include/periph_conf.h (hunk 1 above)
Comment (luisan00) [reply boards/meshme/include/periph_conf.h]: done too

File: tests/periph_adc/doc.txt (hunk 2 above)
Comment (luisan00) [reply tests/periph_adc/doc.txt]: fixed!

**Review by RocioRojas — COMMENTED** [review]
Looks good for me!!!, do a squash

**Comment by luisan00** [comment]
@eduazocar  the proposed changes were made

**Review by eduazocar — APPROVED** [review]

**Pull request merged**